src/clj/web/auth.clj

```diff
@@ -198,19 +198,28 @@
                         :resetPasswordExpires reset-expires}})
     reset-code))
 
+(defn get-user-of-email
+  "Get the user object of an email address from the db"
+  [db email]
+  (mc/find-one-as-map db "users"
+                      {:email email})
+  )
+)
+
 (defn forgot-password-handler
   [{db :system/db
     email-settings :system/email
     {:keys [email]} :params
     headers         :headers}]
   (if (find-non-banned-user db {:email email})
```

Comment:
Instead of querying the db again for the username, I'd make this an `if-let` to bind the `user`.

Reply:
Fixed - hopefully it's correct. I've literally never used Clojure before today.